### PR TITLE
Use sycl::ext::oneapi::experimenta::printf

### PR DESCRIPTION
**Pull request by masterleinad**
This avoids some compiler errors on the testbeds that have already been reported.

### CODE REVIEW

File: benchmarks/dbscan/ArborX_DBSCANVerification.hpp [hunk 1]
```diff
@@ -47,6 +47,9 @@ bool verifyCorePointsNonnegativeIndex(ExecutionSpace const &exec_space,
         {
 #ifndef __SYCL_DEVICE_ONLY__
           printf("Core point is marked as noise: %d [%d]\n", i, labels(i));
+#else
+          sycl::ext::oneapi::experimental::printf(
+              "Core point is marked as noise: %d [%d]\n", i, labels(i));
 #endif
```
Comment (aprokop) [inline benchmarks/dbscan/ArborX_DBSCANVerification.hpp:53]: Maybe do the following way for all of them?
```suggestion
#ifdef __SYCL_DEVICE_ONLY__
          sycl::ext::oneapi::experimental::
#endif
          printf("Core point is marked as noise: %d [%d]\n", i, labels(i));
```

**Comment by dalg24** [comment]
> This avoids some compiler errors on the testbeds that have already been reported.

What compile error?

**Comment by masterleinad** [comment]
> What compile error?

https://github.com/intel/llvm/blob/7589ac9dbb6c85525766b874f894388598211e07/sycl/include/sycl/handler.hpp#L678-L689

**Comment by dalg24** [comment]
Are you saying the current `master` is broken?

**Comment by masterleinad** [comment]
> Are you saying the current `master` is broken?

Yes, with some compiler versions.

**Review by aprokop — APPROVED** [review]
LGTM

**Review by dalg24 — APPROVED** [review]

**Pull request merged**